File: DotNetGotchas/CSharp/ESAutoComplete/PerilsOfAutoCommit/ESLib/MyComp.cs
//MyComp.cs as part of ESLib.dll
using System;
using System.EnterpriseServices;

namespace ESLib
{
	[Transaction(TransactionOption.Required)]
	public class MyComp : ServicedComponent
	{
		private string theMessage = "UnSet";

		[AutoComplete]
		public void SetInfo(string msg)
		{
			theMessage = msg;
		}

		[AutoComplete]
		public string GetInfo()
		{
			return theMessage;
		}
	}
}

File: DotNetGotchas/VBDN/ESAutoComplete/PerilsOfAutoCommit/ESLib/MyComp.vb
'MyComp.vb as part of ESLib.dll
Imports System.EnterpriseServices

<Transaction(TransactionOption.Required)> _
Public Class MyComp
    Inherits ServicedComponent

    Private theMessage As String = "UnSet"

    <AutoComplete()> _
    Public Sub SetInfo(ByVal msg As String)
        theMessage = msg
    End Sub

    <AutoComplete()> _
    Public Function GetInfo() As String
        Return theMessage
    End Function

End Class